numpy/core/src/multiarray/buffer.c

```diff
@@ -596,6 +630,13 @@ _buffer_get_info(PyObject *obj)
             item = PyList_GetItem(item_list, PyList_GET_SIZE(item_list) - 1);
             old_info = (_buffer_info_t*)PyLong_AsVoidPtr(item);
```

Comment:
As an aside, I think we could get a reasonable speedup by using `Py_capsure` objects instead of `PyLong_AsVoidPtr` here - converting pointers to big integers and back seems silly when we could just keep the pointer around.

Reply:
Possibly (and probably nicer in any case).  I would prefer the approach in: gh-16938, but I don't have the mental capacity to figure out how annoying growing the struct is (I doubt its very) and mainly what could be done to reduce the chance of issues.